<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <arg name="world"  default="small_office_inventory"/>
  <param name="world_name" value="$(arg world)"/>
  <arg name="multi" default=""/>
  <arg name="gzpose" default="-x 0.0 -y 0.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/>
  <arg name="gui" default="true"/>
  <arg name="debug" default="false"/>
  <arg name="recording" default="false"/>
  <arg name="extra_gazebo_args" default=""/>
  <arg name="use_moveit_camera" default="false"/>
  <arg name="use_dynamic_footprint" default="false"/>
  <arg name="global_planner" default="global_planner"/>
  
  <!--arguments used by other custom launches -->
  <arg name="map_only" default="false"/>
  <arg name="load_yaml_map" default="true"/>
  <arg name="sim_variant" default="0"/>

  <!--map,localization and transformation for robot/map -->
  <!--needed only when loading custom map.yaml files -->
  <group if="$(arg load_yaml_map)">
    <include file="$(find tiago_2dnav_gazebo)/launch/navigation.launch">
      <arg name="mapping"        value="gmapping"/>
      <arg name="global_planner" value="$(arg global_planner)"/>
      <arg name="local_planner"  value="teb"/>
      <arg name="public_sim"     value="true"/>
      <arg name="config_base_path" value="$(find my_simulation)"/>
      <arg name="map" value="$(find mapping)/maps/$(arg world)"/>
    </include>
  </group>


  <!--tiago gazebo siumlation -->
  <include file="$(find tiago_gazebo)/launch/tiago_gazebo.launch">
	  <arg name="world"  value="$(arg world)"/>
	  <arg name="robot"  value="iron"/>
	  <arg name="arm"  value="false"/>
	  <arg name="end_effector"  value="false"/>
	  <arg name="ft_sensor"  value="false"/>
	  <arg name="multi" value="$(arg multi)"/>
	  <arg name="gzpose" value="$(arg gzpose)"/>
	  <arg name="gui" value="$(arg gui)"/>
	  <arg name="debug" value="$(arg debug)"/>
	  <arg name="public_sim" value="true"/>  <!-- nie uruchamia siÄ™ dla false -->
	  <!--<arg name="tuck_arm" value="true"/>-->
	  <arg name="tuck_arm" value="false"/>
	  <arg name="recording" value="$(arg recording)"/>
	  <arg name="extra_gazebo_args" value="$(arg extra_gazebo_args)"/>
	  <arg name="use_moveit_camera" value="$(arg use_moveit_camera)"/>
	  <arg name="use_dynamic_footprint" value="$(arg use_dynamic_footprint)"/>
  </include>
  
  <!--spawning additional objects in world -->
  <group unless="$(arg map_only)">
    <arg name="yaml" value="$(eval find('my_simulation')+'/params/'+arg('world')+'.yaml')"/>
    <rosparam file="$(arg yaml)" />
    <node pkg="my_simulation" type="spawn.py" name="spawn" output="screen" args="$(arg sim_variant)"/>
  </group>
  
  <!-- pointcloud to laserscan-->
  <include file="$(find tiago_laser_sensors)/launch/rgbd_cloud_laser.launch">
      <arg name="cloud" value="/xtion/depth_registered/points"/>
  </include>


</launch>
